### PR TITLE
Feature. Auto-fixes for null-checks

### LINKED ISSUE #538: Rule 4.3.3: auto-fixes for null-checks

### PR DESCRIPTION
**Pull request by aktsay6**
### What's done:
 * Added fix tests
 * Added fix logic

This pull request closes #538

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt [hunk 1]
```diff
@@ -72,20 +78,77 @@ class NullChecksRule(private val configRules: List<RulesConfig>) : Rule("null-ch
...
+    private fun fixNullInIfCondition(condition: ASTNode,
+                                     binaryExpression: ASTNode,
+                                     isEqualToNull:Boolean) {
+        val variableName = binaryExpression.firstChildNode.text
+        val thenCodeLines = condition
+                .treeParent
+                .getFirstChildWithType(THEN)!!
+                .text
+                .drop(1)
+                .dropLast(1)
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt:105]:
```suggestion
                .trim('{', '}')
```
Is it for this? I think explicitly stating removed characters will be more readable.

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt [hunk 2]
```diff
@@ -72,20 +78,77 @@ class NullChecksRule(private val configRules: List<RulesConfig>) : Rule("null-ch
...
+                .split("\n")
+                .filter { it.isNotBlank() }
+                .map { it.trim() }
+                .toList()
+        val elseCodeLines = condition
+                .treeParent
+                .getFirstChildWithType(ELSE)
+                ?.text
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt:113]: Also, these calls can be extracted to smth like `ASTNode.extractLinesFromBlock()`

**Review by petertrr — CHANGES_REQUESTED** [review]

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt [hunk 3]
```diff
@@ -72,20 +78,77 @@ class NullChecksRule(private val configRules: List<RulesConfig>) : Rule("null-ch
...
+                ?.drop(1)
+                ?.dropLast(1)
+                ?.split("\n")
+                ?.filter { it.isNotBlank() }
+                ?.map { it.trim() }
+                ?.toList()
+        val text = if (isEqualToNull) {
+            if (elseCodeLines.isNullOrEmpty()) {
+                "$variableName ?: kotlin.run {\n${thenCodeLines.joinToString(separator = "\n")}\n}"
```
[the remark this answers is not in the document]
Comment (aktsay6) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt]: Fixed

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt [hunk 4]
```diff
@@ -72,20 +78,77 @@ class NullChecksRule(private val configRules: List<RulesConfig>) : Rule("null-ch
...
+                    |$variableName?.let {
+                    |${thenCodeLines.joinToString(separator = "\n")}
+                    |} ?:
+                    |kotlin.run {
+                    |${elseCodeLines.joinToString(separator = "\n")}
+                    |}
+                """.trimMargin()
+            }
+        }
+        val tree = KotlinParser().createNode(text)
```
[the remark this answers is not in the document]
Comment (aktsay6) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt:126]: Added

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt [hunk 5]
```diff
@@ -72,20 +78,77 @@ class NullChecksRule(private val configRules: List<RulesConfig>) : Rule("null-ch
...
+            }
+        } else {
+            if (elseCodeLines.isNullOrEmpty()) {
+                "$variableName?.let {\n${thenCodeLines.joinToString(separator = "\n")}\n}"
+            } else {
+                """
+                    |$variableName?.let {
+                    |${thenCodeLines.joinToString(separator = "\n")}
+                    |} ?:
+                    |kotlin.run {
```
[the remark this answers is not in the document]
Comment (aktsay6) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt]: Fixed

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt (hunk 1 above)
Comment (aktsay6) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt]: Fixed

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt (hunk 2 above)
Comment (aktsay6) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt]: Fixed

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt [hunk 6]
```diff
@@ -104,17 +146,31 @@ class NullChecksRule(private val configRules: List<RulesConfig>) : Rule("null-ch
                     if (listOf(ElementType.EXCLEQ, ElementType.EXCLEQEQEQ).contains(condition.operationToken)) {
                         warnAndFixOnNullCheck(
                             condition,
-                            false,
+                            true,
                             "use 'requireNotNull' instead of require(${condition.text})"
                         ) {
-                            // todo implement fixer
+                            val variableName = binaryExprNode.firstChildNode.text
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt:152]: Will firstChildNode always be the needed one here? Maybe, `psi.getLeft()` will be more reliable?

**Review by petertrr — APPROVED** [review]

**Pull request merged**